src/Tasks/Microsoft.NET.Build.Tasks/buildCrossTargeting/Microsoft.NET.Sdk.targets

```diff
@@ -30,5 +30,48 @@ Copyright (c) .NET Foundation. All rights reserved.
   <Target Name="Publish">
     <Error Text="The 'Publish' target is not supported without specifying a target framework. The current project targets multiple frameworks, please specify the framework for the published application." />
   </Target>
+
+  <!--
+  ============================================================
+                                      GetAllRuntimeIdentifiers
+
+  Outer build implementation of GetAllRuntimeIdentifiers returns
+  a union of all runtime identifiers used across inner and outer
+  build evaluations.
+
+  It is further set to run before '_GenerateRestoreProjecSpec' 
+  (note that running only 'Restore' is too late and will not work
+  with solution level restore). This ensures that any conditioning
+  of runtime  identifiers against TargetFramework does not prevent
+  restore from providing  the necessary RID-specific assets for all
+  inner builds.
   
+  It also brings parity to VS vs. command line behavior in this
+  scenario because VS passes all of the information from each
+  configured inner build to restore, whereas command-line restore
+  without this target would only use the runtime identifiers that
+  are statically set in the outer evaluation.
+  ============================================================
+  -->
+  <Target Name="GetAllRuntimeIdentifiers" 
+          Returns="$(RuntimeIdentifiers)"
+          BeforeTargets="_GenerateRestoreProjectSpec">
+
+    <ItemGroup>
+      <_GetAllRuntimeIdentifiersTargetFrameworks Include="$(TargetFrameworks)" />
+      <_AllRuntimeIdentifiers Include="$(RuntimeIdentifiers);$(RuntimeIdentifier)" />
+    </ItemGroup>
+
+    <MSBuild Projects="$(MSBuildProjectFile)"
+             Condition="'$(TargetFrameworks)' != ''"
```

Comment:
This condition doesn't seem to be strictly necessary, as this file should only be imported when `TargetFrameworks` is defined, right?